DNN Platform/Library/Entities/Tabs/TabVersions/TabVersionController.cs

```diff
@@ -112,6 +119,34 @@ private static string GetTabVersionsCacheKey(int tabId)
         {
             return string.Format(DataCache.TabVersionsCacheKey, tabId);
         }
+
+        private IEnumerable<TabVersion> ApplyUserLocalTime(IEnumerable<TabVersion> tabVersions, TimeSpan userUtcOffset, TimeZoneInfo userTimeZone)
+        {
+            return tabVersions.Select((tabVersion) =>
+            {
+                var serverUtcOffset = TimeZoneInfo.Local.BaseUtcOffset;
+                if (serverUtcOffset.CompareTo(userUtcOffset) >= 0)
+                {
+                    tabVersion.CreateOnUserLocalDate = ApplyDaylightSavingHour(userTimeZone, tabVersion.CreatedOnDate.Subtract(serverUtcOffset).Add(userUtcOffset));
+                }
+                else
+                {
+                    tabVersion.CreateOnUserLocalDate = ApplyDaylightSavingHour(userTimeZone, tabVersion.CreatedOnDate.Add(serverUtcOffset).Subtract(userUtcOffset));
+                }
+                return tabVersion;
```

Comment:
The currently displayed time is UTC or server time?

Either way you do not have to manually calculate the times. Please check [this guide][tz] for timezone-related APIs. It's a simple built-in call to convert from one timezone to the other.

[tz]: https://docs.microsoft.com/en-us/dotnet/standard/datetime/converting-between-time-zones#converting-between-any-two-time-zones